### PR TITLE
Fix: page metadata

### LINKED ISSUE #254: Page metadata should be specific to the page
Right now we [hardcode the `<meta>` tags](https://github.com/cal-itp/calitp.org/blob/main/src/_includes/meta.html) to the `site` variables, e.g. `site.title` and `site.description`.

This means when links unfurl in e.g. social media or Slack, we always get the default site details instead of the specifics for the link itself (e.g. a Press Release).

All metadata should be specific to the page in question, and only fallback to the `site` configuration when page-level config is unavailable.

## Todo

- [ ] Confirm default page title (browser tabs, link unfurls) for homepage: `Cal-ITP: California Integrated Travel Project`
- [ ] Confirm default page description (link unfurls) for homepage: `A state government initiative, Cal-ITP is making riding by rail and bus simpler and more cost-effective—for California transit providers and riders.`
- [ ] Figure out page title (browser tabs, link unfurls) format for `/press`, `/press/<release>`, and `/resources`
- [ ] Figure out description (link unfurls) for `/press` and `/resources`
- [ ] Confirm using `press.intro` for `/press/<release>` description (link unfurls)

### PR DESCRIPTION
**Pull request by thekaveman**
Use individual page titles/descriptions, update `og` and other `<meta>` tags.

Closes #254

### CODE REVIEW

File: src/_includes/meta.html [hunk 1]
```diff
@@ -1,17 +1,42 @@
...
+    <meta property="article:tag" content="{{ tag }}" />
+  {%- endfor -%}
+  <meta property="article:published_time" content="{{ page.date | date: "%FT%T%z" }}" />
+  <meta property="og:type" content="article">
+{%- else -%}
+  <meta property="og:type" content="website">
+{%- endif -%}
+<meta property="og:locale" content="en_US">
+<meta property="og:site_name" content="{{ site.title }}">
+<meta property="og:title" content="{{ page_title }}" />
```
Comment (angela-tran) [inline src/_includes/meta.html:33]: Do we need an `if page_title` around this?

File: src/_includes/meta.html [hunk 2]
```diff
@@ -1,17 +1,42 @@
...
-<meta property="twitter:domain" content="{{site.domain}}" />
-<meta property="twitter:url" content="{{site.url}}" />
-<meta name="twitter:title" content="{{site.title}}" />
-<meta name="twitter:description" content="{{site.description}}" />
-<meta name="twitter:image" content="{{site.url}}/images/og-image.png" />
+<meta name="twitter:creator" content="@California_ITP">
+<meta name="twitter:description" content="{{description}}" />
+<meta name="twitter:image" content="{{ site.url | absolute_url }}/images/og-image.png" />
+<meta name="twitter:site" content="@California_ITP">
+<meta name="twitter:title" content="{{ page_title }}" />
```
Comment (angela-tran) [inline src/_includes/meta.html:41]: Same here - do we need an `if page_title` around this?

File: src/_includes/meta.html (hunk 1 above)
Comment (thekaveman) [reply src/_includes/meta.html:34]: Good catch. I set `page_title = site.title` if we don't have a `page.title`

File: src/_includes/meta.html (hunk 2 above)
Comment (thekaveman) [reply src/_includes/meta.html:42]: Good catch. I set `page_title = site.title` if we don't have a `page.title`

File: src/_includes/meta.html [hunk 3]
```diff
@@ -1,17 +1,43 @@
...
+{%- if page.title -%}
+  {%- assign page_title = page.title -%}
+  {%- assign title = page.title | append: " - " | append: site.title -%}
+{%- else -%}
+  {%- assign page_title = site.title -%}
+  {%- assign title = site.title -%}
+{%- endif -%}
+
+{%- if page.description -%}
+  {%- assign description = page.description | strip_newlines -%}
```
Comment (angela-tran) [inline src/_includes/meta.html:10]: You might've already seen this same comment in Slack, but putting on the PR too for good measure:

The meta description for the recent ODS press release at least seems to lose spacing due to this `strip_newlines`



`<meta name="description"...>`

**Comment by thekaveman** [comment]
@angela-tran @machikoyasuda I think Slack has a cached version of the site or something, because the metatags on the preview show the description correctly:

image

```html
<meta name="description" content="The California Integrated Travel Project (Cal-ITP) today announced significant progress in implementing the Operational Data Standard (ODS) for transit agencies. In a major milestone for the project, the first vendor-to-vendor integrations powered by ODS have been officially launched. Five companies—EQUANS, Remix (owned by Via), Schedule Masters, Swiftly, and GIRO—now are capable of reading or writing ODS feeds in their software offerings.">
```

**Comment by machikoyasuda** [comment]
image
image
Using this tool: https://socialsharepreview.com/?url=https://deploy-preview-255--cal-itp-website.netlify.app/press/cal-itp-partners-celebrate-major-milestone-implementation-ods

image

This site says there should be `og:description` and `og:title` for LinkedIn.

image

**Comment by machikoyasuda** [comment]
This one has a Discord link tester! https://www.opengraph.xyz/url/https%3A%2F%2Fdeploy-preview-255--cal-itp-website.netlify.app%2Fpress%2Fcal-itp-partners-celebrate-major-milestone-implementation-ods

Maybe we don't need `og:description` and `og:title` cos it looks like it's only for Facebook, which I don't think we care as much about.

**Comment by thekaveman** [comment]
Nice resource @machikoyasuda thanks!

**Comment by thekaveman** [comment]
@machikoyasuda How much do you think we care about those length limitations / suggestions?

**Comment by machikoyasuda** [comment]
@thekaveman I think it's fine if to ignore for now. Slack shows the whole thing.

**Review by machikoyasuda — APPROVED** [review]

**Comment by machikoyasuda** [comment]
@thekaveman The character limit is more of a thing for Google results, Twitter and other social share boxes that are contrained by space. 
image

**Review by angela-tran — APPROVED** [review]

**Pull request merged**